transports/httpclient/src/main/java/org/commonjava/maven/galley/transport/htcli/internal/HttpDownload.java

```diff
@@ -179,6 +158,32 @@ private void writeTarget()
 
                 logger.info( "Closing output stream: {}", out );
                 closeQuietly( out );
+
+                /*
+                * If the server responded with 200 but the content length doesn't match,
+                * we will delete the file and return a null Transfer as if the file wasn't found,
+                * since the content is invalid.
+                */
+                long contentLength = transferSizes.get( target );
+                if ( contentLength >0 )
+                {
+                    if ( contentLength != target.length() )
+                    {
+                        try
+                        {
+                            target.delete();
+                        }
+                        catch ( final Exception e )
+                        {
+                            throw new TransferException(
```

Comment:
You can't throw an exception from a finally clause. If the finally executes after an exception in the try section, then an exception is thrown from the finally section, one of the two will be lost. **This is VERY dangerous.**

If you need to execute this regardless of try / catch, then store the exception from the catch into a variable, then put this logic below the finally section, and make a decision about what to throw...documenting your decision and logging both errors no matter what.

**HOWEVER** I'd rather see this in the try section, and then in the catch clause just call `target.delete()`. If there's an exception that we're throwing, the associated Transfer is automatically invalid IMO.

Reply:
You are right. It is very dangerous , throw exception in finally section.
I remove the code section because of we ready length of target need close the outputstream.
So, I can close it in try block and check content length, then close it again in finally block.
Is it right ?